lib/withOnyx.js

```diff
@@ -67,12 +78,37 @@ export default function (mapOnyxToState) {
                 });
             }
 
+            /**
+             * This method is used externally by sendDataToConnection to prevent unnecessary renders while a component
+             * still in a loading state. The temporary initial state is saved to the component instance and setState()
+             * only called once all the necessary data has been collected.
+             *
+             * @param {String} statePropertyName
+             * @param {*} val
+             */
+            setInitialState(statePropertyName, val) {
+                if (!this.state.loading) {
+                    console.error('withOnyx.setInitialState() called after loading: false');
+                    return;
+                }
```

Comment:
@marcaaron 
Sorry we've made a mistake here 
This method is not called only for initial state but for updates caused by the logic in `componentDidUpdate`: https://github.com/Expensify/react-native-onyx/pull/97/files#diff-364fd7349a1af7b52091427c5a0734da8ff56057fbc7fe2be10734a06c97264cL55

We should update the logic to address calls triggered by updates

```jsx
setInitialState(statePropertyName, val) {
  if (this.state.loading) {
      this.tempState[statePropertyName] = val;

    if (_.some(requiredKeysForInit, key => _.isUndefined(this.tempState[key]))) {
      return;
    }
    
    this.setState({...this.tempState, loading: false});
    delete this.tempState;
  }
  else {
     this.setState({[statePropertyName]: val});
  }
}
```

Also we should maybe change the method name and description to reflect that it's not called only for the initial state

Reply:
Oh I see. The `Onyx.connect()` method may be called again. That solution makes sense to me.

Reply:
Thanks for. catching this one. Created a follow up here -> https://github.com/Expensify/react-native-onyx/pull/98